input/drivers_joypad/switch_joypad.c

```diff
@@ -170,49 +170,117 @@ static void switch_joypad_destroy(void)
 }
 
 #ifdef HAVE_LIBNX
-int lastMode = 0; /* 0 = handheld, 1 = whatever */
+int previousHandheld = -1; /* 1 = handheld, 0 = docked, -1 = first use */
+uint previousSplitJoyconSetting[MAX_USERS] = { 0 };
 static void switch_joypad_poll(void)
 {
-   settings_t *settings = config_get_ptr();
+   settings_t *settings;
+   int i;
+   int handheld;
 
+   settings = config_get_ptr();
    hidScanInput();
-
-   if (!hidGetHandheldMode())
+   handheld = hidGetHandheldMode();
+   
+   if (previousHandheld == -1)
    {
-      if (lastMode != 1)
-      {
-         unsigned i = 0;
+      /* first call of this function, apply joycon settings according to prefs and init vars*/
+      if (!handheld) {
          for(i = 0; i < MAX_USERS; i += 2)
          {
-            /* CONTROLLER_PLAYER_X, X == i++ */
             if (settings->uints.input_split_joycon[i])
             {
                hidSetNpadJoyAssignmentModeSingleByDefault(i);
                hidSetNpadJoyAssignmentModeSingleByDefault(i + 1);
                hidSetNpadJoyHoldType(HidJoyHoldType_Horizontal);
+            } 
+            else if (!settings->uints.input_split_joycon[i])
+            {
+               hidSetNpadJoyAssignmentModeDual(i);
+               hidSetNpadJoyAssignmentModeDual(i + 1);
+               hidMergeSingleJoyAsDualJoy(i, i + 1);
             }
          }
-         lastMode = 1;
+      }
+      previousHandheld = handheld;
+      for(i = 0; i < MAX_USERS; i += 2)
+      {
```

Comment:
If there is only one line for a code block, then just get rid of the brackets.

Reply:
Thanks! Fixed in latest force-push.